### PR TITLE
Fate#318832 Redesign iSCSI Client/Server UI - Authentication

### PR DESCRIPTION
**Pull request by ls-zhu**
Fate#318832 Redesign iSCSI Client/Server UI - Authentication.

Add two functions: EnableDiscoveryAuth and DisableDiscoveryAuth
A bug fix for discovery session without authentication
Changed online help docs

### CODE REVIEW

File: src/include/iscsi-lio-server/helps.rb [hunk 1]
```diff
@@ -33,6 +33,10 @@ def initialize_iscsi_lio_server_helps(include_target)
       textdomain "iscsi-lio-server"
 
       # All helps are here
+      use_login_auth = "If you enable <b>Use Login Authentication</b>, you need to fill in <b>UserID</b> and  <b>Password</b> for both \
```
Comment (jreidinger) [inline src/include/iscsi-lio-server/helps.rb]: not translated

File: src/include/iscsi-lio-server/helps.rb [hunk 2]
```diff
@@ -155,6 +161,9 @@ def initialize_iscsi_lio_server_helps(include_target)
             "<p>Under <b>Ip Address</b> and <b>Port Number</b> you specify under which address\n" +
               "and port the service will be available. Default for port number is 3260.\n" +
               "Only ip addresses assigned to one of the network cards are possible."
+          ) + "</p>" +"<p>" +
+          _(
```
Comment (jreidinger) [inline src/include/iscsi-lio-server/helps.rb]: this won't work. you need to translate it above. another way is to mark string for translation with `N_()` and then use `_()` here.

File: src/include/iscsi-lio-server/widgets.rb [hunk 3]
```diff
@@ -188,9 +188,21 @@ def validateGlobal(key, event)
...
+        IscsiLioData.DisableDiscoveryAuth()
+      else
+        IscsiLioData.EnableDiscoveryAuth()
+      end
+    end
     # handle authentication dialog
     def handleAuth(key, event)
       event = deep_copy(event)
+      # we need to call handleDiscAuthStatus() here because we must handle pre set status of discovery authentication 
+      # like "disable" for defaulf.
```
Comment (jreidinger) [inline src/include/iscsi-lio-server/widgets.rb]: typo `default`

File: src/include/iscsi-lio-server/widgets.rb [hunk 4]
```diff
@@ -199,6 +211,9 @@ def handleAuth(key, event)
             status = Convert.to_boolean(UI.QueryWidget(Id(:auth_none), :Value))
             SetAuthIn(!status)
             SetAuthOut(!status)
+            # we must also call handleDiscAuthStatus() here because once users enable / disable discovery authentication,
+            # we need to set the configfs attibutes  accrodingly.
```
Comment (jreidinger) [inline src/include/iscsi-lio-server/widgets.rb]: typo `configs` and `accordingly`

File: src/modules/IscsiLioData.rb [hunk 5]
```diff
@@ -83,6 +83,16 @@ def SetTgt(data, tgt, tpg, m)
...
 
+    def EnableDiscoveryAuth()
+        cmd ="lio_node --setchapdiscenforce=1"
+        ret = LogExecCmd("#{cmd}", do_log: true)
+    end
+
+    def DisableDiscoveryAuth()
+      cmd ="lio_node --setchapdiscenforce=0"
+      ret = LogExecCmd("#{cmd}", do_log: true)
+    end
```
Comment (jreidinger) [inline src/modules/IscsiLioData.rb]: it is almost identical values. Why not use `discovery_auth=` method like
```
def discovery_auth=(enable)
   cmd ="lio_node --setchapdiscenforce=#{enable ? 1 : 0}"
   ret = LogExecCmd(cmd, do_log: true)
end
```

**Comment by ls-zhu** [comment]
will submit another pull request now

**Pull request closed without merging**